src/main/java/org/jnode/fs/xfs/XfsFileSystem.java

```diff
@@ -48,14 +74,31 @@ public XfsFileSystem(Device device, FileSystemType<? extends FileSystem<XfsEntry
         super(device, true, type);
     }
 
+    public static String HexToAscii(String hexString) {
+        try {
+            StringBuilder ascii = new StringBuilder();
+            final Iterable<String> chars = Splitter.fixedLength(2).split(hexString);
+            for (String c : chars) {
+                ascii.append((char) Byte.parseByte(c, 16));
+            }
+            return ascii.toString();
+        } catch (Throwable t) {
+            return "INVALID";
+        }
+    }
+
     /**
      * Reads in the file system from the block device.
      *
      * @throws FileSystemException if an error occurs reading the file system.
      */
-    public final void read() throws FileSystemException {
+    public final void read() throws FileSystemException, IOException {
         superblock = new Superblock(this);
         agINode = new AllocationGroupINode(this);
+        iNodeSize = superblock.getInodeSize();
+        blockSize = superblock.getBlockSize();
+        aGCount = (int) superblock.getAGCount();
+        allocationGroupSize = blockSize * superblock.getTotalBlocks() / aGCount;
```

Comment:
It would seem that the thrown exception `IOException` is caused by this line. Instead of throwing IOException, you could put this line in the `try` block below, but after reviewing the `getTotalBlocks()` function you could remove the thrown `IOException` in the `getTotalBlocks()` function declaration, as `IOException` is not thrown there. Then you could revert the need to catch `IOException` here and the calling class.

Reply:
code change applied. Thanks.